wpimath/src/main/native/include/frc/MathUtil.h

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+#pragma once
+
+#include <wpi/math>
+
+#include "units/angle.h"
+
+namespace frc {
+
+/**
+ * Returns modulus of input.
+ *
+ * @param input        Input value to wrap.
+ * @param minimumInput The minimum value expected from the input.
+ * @param maximumInput The maximum value expected from the input.
+ */
+template <typename T>
+constexpr T InputModulus(T input, T minimumInput, T maximumInput) {
```

Comment:
I'm not sure of the mathematical soundness of this... it's not quite the same as [modulo with offset](https://en.wikipedia.org/wiki/Modulo_operation#Modulo_with_offset).  What does this operation do when min and max are unbalanced or not centered at zero (e.g. min is 1 and max is 3).. what does an input of 0 or 4 result in in that case?

Reply:
The modulo with offset math on that Wikipedia page doesn't give the correct answer. Consider an input of -340 with a range of -180 to 180.

```
From Wikipedia's notation:
d <= a <= d + n - 1

minInput <= input <= maxInput

minInput = d
input = a
maxInput = d + n - 1
maxInput = minInput + n - 1
n - 1 = maxInput - minInput
n = maxInput - minInput + 1

a mod_d n = a - n * std::floor((a - d) / n)


Let input = -340, min = -180, and max = 180.

mod = max - min + 1
    = 180 - (-180) + 1
    = 361
floor = (int) (input - min) / mod
      = (int) (-340 - (-180)) / 361
      = (int) -160 / 361
      = (int) -0.44
      = 0
return input - mod * floor
return -340 - 361 * 0
return -340

```

The answer should be 20 to keep it within the range. Even if the floor function gave the correct result, the multiplier on it is off by one.